pkg/start/start.go

```diff
@@ -133,19 +135,28 @@ func (o *Options) Run(ctx context.Context) error {
 	// check to see if techpreview should be on or off.  If we cannot read the featuregate for any reason, it is assumed
 	// to be off.  If this value changes, the binary will shutdown and expect the pod lifecycle to restart it.
 	startingFeatureSet := ""
-	gate, err := cb.ClientOrDie("feature-gate-getter").ConfigV1().FeatureGates().Get(ctx, "cluster", metav1.GetOptions{})
-	switch {
-	case apierrors.IsNotFound(err):
-		// if we have no featuregates, then the cluster is using the default featureset, which is "".
-		// This excludes everything that could possibly depend on a different feature set.
-		startingFeatureSet = ""
-	case err != nil:
-		// client-go automatically retries network blip errors on GETs for 30s by default.  If we fail longer than that
-		// the operator won't be able to do work anyway.  Return the error and crashloop.
-		return err
 
-	default:
-		startingFeatureSet = string(gate.Spec.FeatureSet)
+	if err := wait.PollImmediate(2*time.Second, 30*time.Second, func() (done bool, err error) {
+		gate, err := cb.ClientOrDie("feature-gate-getter").ConfigV1().FeatureGates().Get(ctx, "cluster", metav1.GetOptions{})
+		switch {
+		case apierrors.IsNotFound(err):
+			// if we have no featuregates, then the cluster is using the default featureset, which is "".
+			// This excludes everything that could possibly depend on a different feature set.
+			startingFeatureSet = ""
+			return true, nil
+			// retry some network blips that client-go does not, to survive brief API disruptions during install
+		case net.IsTimeout(err) || net.IsConnectionRefused(err):
+			return false, nil
+		case err != nil:
+			// client-go automatically retries some network blip errors on GETs for 30s by default.  If we fail longer than that
+			// the operator won't be able to do work anyway.  Return the error and crashloop.
+			return false, err
```

Comment:
We might want to log `err`, store in a new `lastError`, and then return `nil` here.  Because [`ConditionFunc`][1] has:

> ConditionFunc returns true if the condition is satisfied, or an error if the loop should be aborted.

and I thought the idea was that until we reach 30s, we want to keep trying.

[1]: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ConditionFunc

Reply:
addressed in 296ae5f8a950fbda3e348dd131bde3d3a95f92ce